keps/sig-network/1111-dynamically-re-configure-lb-node-set/README.md

```diff
@@ -0,0 +1,681 @@
+itself.  
+
+### Goals
+
+#### Defining "healthy"
+
+- Answer the question: what constitutes a "healthy" node from the point of view
+  of an __application load balancer__? Under which conditions should a node be /
+  not be considered as an acceptable next hop for load balancer traffic? Should
+  a healthy __node__ even be the guiding criteria determining where an
+  __application__ load balancer should send traffic?
+
+- Improve the performance of keeping the cloud provider load balancers in sync
+  with the state of the Kubernetes cluster, by dynamically having the load
+  balancer identify the set of nodes to be used, solely through the utilization
+  of their health check (HC) probes.  
+
+#### Align health checks between services
+
+- Reduce service downtime caused by updating fields on the service object which
+  do not have a direct correlation to the load balancer but more so to its HC /
+  where traffic should be forwarded, ex: updating the `externalTrafficPolicy`
+  field from `Cluster` to `Local` or vice versa. Such changes to the service
+  object should not end up re-configuring the load balancer, only change the
+  response returned by the service proxy to the load balancer's HC.
+
+```
+<<[UNRESOLVED HC-linked-service-fields ]>>
+what are all the fields on the service object which fulfill this statement?
+<<[/UNRESOLVED]>>
+```
+
+### Non-Goals
+
+- Avoiding re-configuration of the load balancer for fully deleted / newly added
+  cluster nodes.
+
+## Proposal
+
+Up until this point the KCCM has defined the following conditions for what is
+considered a "healthy" node:
+
+1. Is the node not tainted with the key: `ToBeDeletedByClusterAutoscaler`?
+2. Is the node condition `NodeReady` equal to `ConditionTrue`?
+
+This first predicate was introduced by the
+[Kube PR #105946](https://github.com/kubernetes/kubernetes/pull/105946) which
+stated:
+
+> If multiple...nodes are being deleted, the list of nodes passed to
+> cloud-provider code via UpdateLoadBalancer() will be stale, causing a few
+> failed updates before the backends stabilize.
+
+It seemed, from reading the above, that the goal was to ensure that we get a
+head start at deleting the node from the load balancer set before the underlying
+VM instance has been completely cleaned up by the autoscaler. The interpretation
+of the PR indicated that if the predicate is not used: we'll only get notified
+the moment the node is deleted in the API server, which might be after the VM
+instance is GC'ed, causing the update call on the load balancer to fail because
+the node cannot be found - and thus, possibly, stale references by the load
+balancer to nodes which don't exists anymore.
+
+This cannot be true today however. This is easily logically verifiable by simply
+looking at the generic cloud-provider API for load balancer operations that the
+KCCM can invoke: the only functions which ends up getting called when a node is
```

Comment:
Is this happening with external cloud providers too?

This has to be the same for all cloud providers, not only for internals

Reply:
internals the ones that have the code in kubernetes/kubernetes, external the others